pkgs/build-support/replace-vars/default.nix

```
@@ -0,0 +1,77 @@
+{ lib, stdenvNoCC }:
+
+/**
+  `replaceVars` is a wrapper around the [bash function `substitute`](https://nixos.org/manual/nixpkgs/stable/#fun-substitute)
+  in the stdenv. It allows for terse replacement of names in the specified path, while checking
+  for common mistakes such as naming a substitution that does nothing or forgetting a substitution.
+
+  As with the [`--subst-var-by`](https://nixos.org/manual/nixpkgs/stable/#fun-substitute-subst-var-by)
+  flag, names are encoded as `@name@` in the provided file at the provided path.
+
+  Any unmatched variable names in the file at the provided path will cause a build failure.
+
+  Any remaining text that matches `@[A-Za-z_][0-9A-Za-z_'-]@` in the output after replacement
+  has occurred will cause a build failure.
+
+  Attribute arguments:
+
+  - `path`: A [string-like value](https://nixos.org/manual/nixpkgs/stable/#function-library-lib.strings.isStringLike) of a [store path](https://nixos.org/manual/nix/stable/store/store-path.html#store-path).
+
+  - `attrs`: each entry in this set corresponds to a `--subst-var-by` entry in `substitute`.
+    See https://nixos.org/manual/nixpkgs/stable/#fun-substitute
```

```suggestion
  # Inputs

  `path` ([Store Path](https://nixos.org/manual/nix/latest/store/store-path.html#store-path) String)
  : The file to substitute values in.

  `attrs` (AttrsOf String)
  : Each entry in this set corresponds to a `--subst-var-by` entry in [`substitute`](https://nixos.org/manual/nixpkgs/stable/#fun-substitute).
```

Using the [documentation conventions](https://github.com/NixOS/nixpkgs/tree/master/doc#documentation-conventions) and some adjustments:
- Don't link to `isStringLike`, I don't think it's helpful here.
- Use the `latest` Nix version link, the `stable` one doesn't seem to exist
- Give a description for `path`
- More brief linking to `substitute`

---

Just noticed that you took the original text here [from me](https://github.com/NixOS/nixpkgs/pull/301350#discussion_r1557887949) :see_no_evil: :sweat_smile:

---

Yep, it's past Silvan talking to now Silvan here 😉 I'll apply the edits.